### DISCUSSION TITLE
AsyncFrameError: Expression evaluation in async frames is not supported

### DISCUSSION THREAD

**Discussion by ramanraja** [firebase/flutterfire#11913] [Q&A]
I have a Flutter application that uses FirebaseAuth. I encountered this error:

```
AsyncFrameError: Expression evaluation in async frames is not supported. No frame with index 29.
#0      DartDebugAdapter.evaluateRequest (package:dds/src/dap/adapters/dart.dart:997:7)
```

The AsyncFrameError is thrown at the line:
`FirebaseAuth.instance.signInWithEmailAndPassword()`
and the execution hits an internal breakpoint and pauses at that line.
 
Minimalistic code to reproduce the above error:

 ```
 Future<bool> login() async {
    print('Logging in..');
    try {
      await FirebaseAuth.instance.signInWithEmailAndPassword(
        email: 'user@example.com',
        password: 'password123',
      );
    } on FirebaseAuthException catch (ae) {
      print(ae.toString());
      return false;
    }
    print('You are now logged in.');
    return true;
  }

  void crashIt() async {
    final result = await login();
    throw ('Deliberate exception');  // normally, exception should be thrown at this line
    print(result);
  }

```
Explanation: The calling function `crashIt()` is async. It deliberately raises an exception.

From within `crashIt()`, FirebaseAuth method `signInWithEmailAndPassword()` is invoked through another async function call. This should normally throw the exception at the exact line containing the `throw()`. But it stops execution at the line `signInWithEmailAndPassword()` and raises the error 'Expression evaluation in async frames is not supported'.

As another experiment, remove the async keyword and modify the calling function to be synchronous. Now the AsyncFrameError error vanishes; the exception is thrown from the correct place.

```
  void crashIt() {
    final result = login(); // no async, no await
    throw ('Deliberate exception'); // now exception is raised at this line, as expected
    print(result);
  }
```
Can someone please explain this behavior?

**Comment by maseappsltd** [comment]
Same error, noone?
